server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -158,6 +158,9 @@ public JacksonSmileProvider getJacksonSmileProvider(@Smile ObjectMapper objectMa
 
   static Server makeJettyServer(DruidNode node, ServerConfig config)
   {
+    if (node.getPort() < 0) {
```

Comment:
I don't think this is the right way to approach this. This makes the `port` control a lot of parameters other than just the port.